### DISCUSSION TITLE
SolarWinds API

### DISCUSSION THREAD

**Discussion by skinfrakkiirid** [grafana/grafana-infinity-datasource#639] [Specific APIs] [answered]
Trying to set up a connection to the Orion SolarWinds API. The newer plugin doesn't have a URL. How would I set up the URL and authentication for this connection?

http://server:17778/SolarWinds/InformationService/v3/Json/

And what allowed hosts do I need? I don't see that in the current docs

Error message:

Status: 500. Message: error getting data frame. requested URL is not allowed. To allow this URL, update the datasource config Security -> Allowed Hosts section

**Comment by yesoreyeram** [comment]
http://server:17778 is the allowed host you need to add in your case.

**Comment by skinfrakkiirid** [comment]
So how would I query SolarWinds with the API URL they give?

**Comment by skinfrakkiirid** [comment]
Status: 500. Message: error getting data frame. error getting response from url http://server:17778/SolarWinds/InformationService/v3/Json/. no response received. Error: Get "http://server:17778/SolarWinds/InformationService/v3/Json/": context deadline exceeded (Client.Timeout exceeded while awaiting headers)

**Comment by skinfrakkiirid** [comment]
Status: 500. Message: error getting data frame. error getting response from url http://server:17774/SolarWinds/InformationService/v3/Json/. no response received. Error: Get "http://server:17774/SolarWinds/InformationService/v3/Json/http://10.1.28.155:17774/SolarWinds/InformationService/v3/Json/": context deadline exceeded (Client.Timeout exceeded while awaiting headers)

**Reply by yesoreyeram** [reply]
Hiya.. Sorry for the delay. It looks like you are adding URL in both settings and query. Instead I would recommend to use `http://server:17774` as the URL in settings and then `/SolarWinds/InformationService/v3/Json/` as the URL in query.

**Comment by societymartingale** [comment]
Just following up, was anyone able to get this working?

**Comment by skinfrakkiirid** [comment]
Haven't got an update

**Comment by Mesverrum** [comment, marked as answer]
I'm pretty late to the party, but I posted up an explanation of what needs to happen to connect Infinity to Orion
https://github.com/grafana/grafana-infinity-datasource/discussions/1137
